src/asammdf/blocks/mdf_v4.py

```diff
@@ -3042,30 +3129,30 @@ def append(
                         )
 
                     byte_size = 8
-                    kwargs = {
+                    cn_kwargs = {
                         "channel_type": v4c.CHANNEL_TYPE_VLSD,
                         "bit_count": 64,
                         "byte_offset": offset,
                         "bit_offset": 0,
                         "data_type": s_type,
-                        "data_block_addr": data_addr,
                         "flags": 0,
                     }
 
                     if invalidation_bytes_nr and signal.invalidation_bits is not None:
                         if (origin := signal.invalidation_bits.origin) == InvalidationArray.ORIGIN_UNKNOWN:
-                            inval_bits[origin].append(signal.invalidation_bits)
+                            invalidation_arrays = typing.cast(list[InvalidationArray], inval_bits[origin])
+                            invalidation_arrays.append(signal.invalidation_bits)
                         else:
                             inval_bits[origin] = signal.invalidation_bits
-                        kwargs["flags"] = v4c.FLAG_CN_INVALIDATION_PRESENT
-                        kwargs["pos_invalidation_bit"] = origin
+                        cn_kwargs["flags"] = v4c.FLAG_CN_INVALIDATION_PRESENT
+                        cn_kwargs["pos_invalidation_bit"] = origin[1]
```

Comment:
nice catch!

Reply:
It's mypy, not me, but I am taking all the credits :D